pyquil/quil.py

```diff
@@ -372,7 +373,7 @@ def while_do(self, classical_reg, q_program):
         self.inst(JumpTarget(label_end))
         return self
 
-    def if_then(self, classical_reg, if_program, else_program=None):
+    def if_then(self, classical_reg: int, if_program: 'Program', else_program: Optional['Program'] = None) -> 'Program':
```

Comment:
Likewise here, if `classical_reg` is an `int` it should raise an exception. For some reason we call `unpack_classical_reg` here but not in `while_do` above, so `classical_reg` can be any `MemoryReferenceDesignator` here, rather than requiring an actual `MemoryReference`.  Probably we should update `while_do` to also call `unpack_classical_reg` for consistency.